File: 1-Iniciante/ex1021.c
#include <stdio.h>

void partInt(int vet[], int x);
void partDec(int vet[], int x);
void imprime(int vetCed[], int vetMoe[]);

int main(){
    /*variaveis*/
    int i, inteira, decimal, cedulas[7] = {0,0,0,0,0,0,0}; 
    int moedas[5] = {0,0,0,0,0};
    double valor;

    /*entrada*/
    scanf("%lf", &valor);

    /*processamento*/
    inteira = (int)(valor);
    decimal = (valor - inteira) * 100;

    partInt(cedulas, inteira);
    partDec(moedas, decimal);

    /*saida*/
    imprime(cedulas, moedas);
    return 0;
}

void partInt(int vet[], int x){
    int i = 0, aux, cedulas[7] = {100, 50, 20, 10, 5, 2, 1};
    while(x > 0){
        if(x >= cedulas[i]){
            vet[i] = x / cedulas[i]; 
            aux = x % cedulas[i];
            x = aux; 
        }
        i++;
    }
}

void partDec(int vet[], int x){
    int i = 0, aux, moedas[5] = {50, 25, 10, 5, 1};
    while(x > 0){
        if(x >= moedas[i]){
            vet[i] = x / moedas[i]; 
            aux = x % moedas[i];
            x = aux; 
        }
        i++;
    }
}

void imprime(int vetCed[], int vetMoe[]){
    int i;
    float cedulas[7] = {100, 50, 20, 10, 5, 2, 1};
    float moedas[5] = {0.5, 0.25, 0.10, 0.05, 0.01};

    printf("NOTAS:\n");
    for(i = 0; i < 6; i++){
        printf("%d nota(s) de R$ %.2f\n", vetCed[i], cedulas[i]);
    }

    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ %.2f\n", vetCed[6], cedulas[6]);

    for(i = 0; i < 5; i++){
        printf("%d moeda(s) de R$ %.2f\n", vetMoe[i], moedas[i]);
    }
}

File: 1-Iniciante/ex1021.js
// entrada
let input = require('fs').readFileSync('dev/stdin', 'utf-8'); //adicionar uma barra antes do 'dev' para funcionar

// processamento & saida
let parte_inteira = parseInt(input.split('.')[0]);
let parte_decimal = parseInt(input.split('.')[1]);

// notas
console.log('NOTAS:');
console.log(`${Math.floor(parte_inteira / 100)} nota(s) de R$ 100.00`);
parte_inteira %= 100;
console.log(`${Math.floor(parte_inteira / 50)} nota(s) de R$ 50.00`);
parte_inteira %= 50;
console.log(`${Math.floor(parte_inteira / 20)} nota(s) de R$ 20.00`);
parte_inteira %= 20;
console.log(`${Math.floor(parte_inteira / 10)} nota(s) de R$ 10.00`);
parte_inteira %= 10;
console.log(`${Math.floor(parte_inteira / 5)} nota(s) de R$ 5.00`);
parte_inteira %= 5;
console.log(`${Math.floor(parte_inteira / 2)} nota(s) de R$ 2.00`);
parte_inteira %= 2;

// moedas
parte_decimal += (parte_inteira * 100);
console.log('MOEDAS:');
console.log(`${Math.floor(parte_decimal / 100)} moeda(s) de R$ 1.00`);
parte_decimal %= 100;
console.log(`${Math.floor(parte_decimal / 50)} moeda(s) de R$ 0.50`);
parte_decimal %= 50;
console.log(`${Math.floor(parte_decimal / 25)} moeda(s) de R$ 0.25`);
parte_decimal %= 25;
console.log(`${Math.floor(parte_decimal / 10)} moeda(s) de R$ 0.10`);
parte_decimal %= 10;
console.log(`${Math.floor(parte_decimal / 5)} moeda(s) de R$ 0.05`);
parte_decimal %= 5;
console.log(`${parte_decimal} moeda(s) de R$ 0.01`);